shared/rakefiles/secrets.rb

```diff
@@ -242,12 +242,18 @@ def self.rotate_key(encryption_key)
 
     begin
       new_version = JSON.parse(new_version)
-      new_version_id = new_version["name"].match(/\/([0-9]+)$/)[1]
+      new_version_id = get_crypto_key_version(new_version["name"])
     rescue
+      puts new_version
       puts "ERROR: Unable to create new version for key '#{encryption_key}', terminating!"
       raise
     end
 
+    return new_version_id
+  end
+
+  # This method disables all versions except primary for target encryption_key
+  def self.disable_key_versions(encryption_key, primary_version_id = 0)
```

Comment:
I think `primary_version_id` should be required, not optional. I don't think it makes sense to call this method without a specific version id.

Reply:
It makes sense to call this method without specific version id if you want to disable kms key entirely. But yeah, yagni this!